api/aether-top-level-openapi3.yaml

```diff
@@ -134,76 +134,265 @@ components:
+                            - ALREADY_EXISTS
+                            - UNAUTHORIZED
+                            - FORBIDDEN
+                            - CONFLICT
+                            - INVALID
+                            - UNAVAILABLE
+                            - NOT_SUPPORTED
+                            - TIMEOUT
+                            - INTERNAL
+                        description:
+                          type: string
+                abort:
+                  description:
+                  properties:
+                    status:
+                      properties:
+                        start:
+                          type: string
+                          format: date-time
+                        end:
+                          type: string
+                          format: date-time
+                    state:
+                      type: string
+                      enum:
+                        - ABORTING
+                        - ABORTED
+            proposals:
+              description: the set of proposals managed by the transaction
+              type: array
             state:
-              description: the state of the transaction within a Phase
+              description: the overall transaction state
               type: string
               enum:
-                - TRANSACTION_PENDING
-                - TRANSACTION_COMPLETE
-                - TRANSACTION_FAILED
-                - TRANSACTION_VALIDATING
-                - TRANSACTION_VALIDATED
-                - TRANSACTION_VALIDATION_FAILED
-          required:
-            - phase
-            - state
-          type: object
-        created:
-          description: the time at which the transaction was created
-          type: string
-          format: date-time
-        updated:
-          description: the time at which the transaction was last updated
-          type: string
-          format: date-time
-        changes:
-          description: a set of changes to apply to targets
-          items:
-            $ref: '#/components/schemas/Change'
-          type: array
-          x-keys: target_id
-        deleted:
-          description: a flag indicating whether this transaction is being deleted by a snapshot
-          type: boolean
-        dependency:
-          description: a reference to the transaction on which this transaction is dependent
-          type: string
-        dependents:
-          description: a list of references to transactions that depend on this transaction
-          items:
-            properties:
-              id:
-                type: string
-            type: object
-          type: array
-          x-keys: id
-        username:
-          description: the name of the user that made the transaction
-          type: string
+                - PENDING
+                - VALIDATED
+                - COMMITTED
+                - APPLIED
+                - FAILED
+            failure:
```

Comment:
I think you could create `Failure` as its own type and ref it here and anywhere else you use Failure

Reply:
Done